### DISCUSSION TITLE
Setting per-user permissions for Samba shares from GUI

### DISCUSSION THREAD

**Discussion by shalak** [45Drives/cockpit-file-sharing#122] [Q&A]
The feature list mentions:

> Manage allow-list of valid users and groups

However, later it's crossed out:

>* ~~Valid Users/Groups~~ removed in v4 - valid users property still available through advanced settings.
>    * ~~Allow-lists for users and groups~~
>    * ~~By default, any user or group is allowed~~

So I understand that before v4 we were able to click and select which user can read and/or write and now we have to manually set the `valid users`, `write list` & `read list` smb.conf params using the "advanced settings" text area? Is my understanding correct?

If so, why was the decision to remove this from GUI made? Ability to easily select user access from GUI (and have it automatically populate those three smb.conf fields) is a great feature - and make it so Cockpit can be basically used as a perfect, light-weight NAS for deployments where you manage the hardware side/ZFS pools on the VM/LXC hypervision level, not NAS software itself. Why was it removed?

**Comment by Crypto-Spartan** [comment]
I'm also wondering why they did this, user permissions was my primary reason for using cockpit + the 45Drives file sharing plugin to manage my samba & nfs shares

additionally, is there a guide on how to do this via advanced settings?

**Comment by rstoll21** [comment]
Only alternative in my mind is [webmin](https://webmin.com/), but that's a lot more than I need. It's realy a pity that they removed the Valid User/Groups feature.
